game_board/avl/avl_handler.py

```diff
@@ -128,8 +128,9 @@ def delNode(self, key, b=True):
 		
 	def delNodeByID(self, nid, b=True):
 		""" remove node from tree by value """
-		self.root = self.tree.delete_node_id(self.root, nid, balance=b)
-		self.balanced = self.tree.isBalanced(self.root)		
+		if nid != self.golden_id:
+			self.root = self.tree.delete_node_id(self.root, nid, balance=b)
+			self.balanced = self.tree.isBalanced(self.root)		
 			
```

Comment:
Added a check to make sure id being removed is not golden. I know the API will not attempt to remove the golden node but it's an extra precaution

Reply:
Extra precautions are good